packages/core/src/Resource/Resource.js

```diff
@@ -163,12 +179,36 @@ Resource.create = (id = null, type, data = null, relationships = null, meta = nu
  *
  * @returns {HyralResource}
  */
-Resource.fromState = (id, type, state) => Resource.create(
-  id,
-  type,
-  state.data || null,
-  state.relationships || null,
-  state.meta || null,
-);
+Resource.fromState = (id, type, state) => {
+  const relationData = mapValues(state.relationships, (relation, relationKey) => {
+    if (!state.data[relationKey]) {
+      return null;
+    }
+
+    if (Array.isArray(state.data[relationKey])) {
+      return state.data[relationKey].map(value => Resource.fromState(
+        state.data[relationKey].id,
+        relation.resource,
+        value || null,
```

Comment:
missing relationships and meta in constructor

Reply:
dat is omdaat fromState alleen maar id, type en state accepteert en relationship en meta staat in de state zoals je kunt zien bij 209 en 210

Reply:
Dat is helemaal waar, ik zag even verkeerd dat deze ook fromState gebruikt.